code/modules/projectiles/updated_projectiles/ammo_datums.dm

```diff
@@ -956,6 +956,27 @@
 	smoke.set_up(1, get_turf(P))
 	smoke.start()
 
+/datum/ammo/rocket/frag
+	name = "fragmentation rocket"
+	hud_state = "rocket_frag"
+
+/datum/ammo/rocket/on_hit_mob(mob/M, obj/item/projectile/P)
```

Comment:
You're re-defining the same proc that's already defined above, perhap you mean
`/datum/ammo/rocket/frag/on_hit_mob(mob/M, obj/item/projectile/P)` instead of `/datum/ammo/rocket/on_hit_mob(mob/M, obj/item/projectile/P)`?